lib/resty/kafka/producer.lua

```diff
@@ -83,20 +83,30 @@ end
 local function produce_decode(resp)
     local topic_num = resp:int32()
     local ret = new_tab(0, topic_num)
+    local api_version = resp.api_version
 
     for i = 1, topic_num do
         local topic = resp:string()
         local partition_num = resp:int32()
 
         ret[topic] = {}
 
+        -- ignore ThrottleTime
         for j = 1, partition_num do
             local partition = resp:int32()
 
-            ret[topic][partition] = {
-                errcode = resp:int16(),
-                offset = resp:int64(),
-            }
+            if api_version == request.API_VERSION_V0 or api_version == request.API_VERSION_V1 then
```

Comment:
better use local variables here, `API_VERSION_V0` and `API_VERSION_V1`, it's acceptable for duplicated code.